api/src/main/java/org/openmrs/module/rulesengine/engine/RulesEngineImpl.java

```diff
@@ -21,7 +20,7 @@
 import java.util.List;
 import java.util.Set;
 
-@Component
```

Comment:
there should be a corresponding bean def in the applicationContext, no?

Reply:
@angshu throwing below exception 
_Annotation-specified bean name 'rulesEngine' for bean class conflicts with existing, non-compatible bean definition of same name and class._
as we are creating object in the below file.
https://github.com/Bahmni/openmrs-module-rulesengine/blob/master/api/src/main/java/org/openmrs/module/rulesengine/rule/CSVBasedDosageRule.java#L33